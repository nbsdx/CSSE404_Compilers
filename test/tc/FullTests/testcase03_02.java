class Test2 {
	public static void main(String[] args) {
		if((3 + 2 * 7) && true) {
			int x = 5;
		} else {
			System.out.println(true);
		}
	}
}

/* This program contains the following typechecking errors:
	The expression 3 + 2 * 7 doesn't evaluate to a boolean
	The argument to System.out.println must be an integer 
*/