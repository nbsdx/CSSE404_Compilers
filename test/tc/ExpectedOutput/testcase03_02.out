Left argument of type int does not match expected type boolean for operator AND
In MiniJava, System.out.println only takes an int.  The expression has type boolean
